### PR TITLE
feat(vscode): Validate Workspace command (#442)

### LINKED ISSUE #442: VS Code extension: palette command to validate all Stagebook files in workspace
Depends on #439.

## Context

The VS Code extension validates files on edit (live diagnostics in the Problems panel), but the Problems panel only shows diagnostics for files VS Code has *opened* in the current session. A treatment author may have a directory of 30 `.stagebook.yaml` files; the extension only flags issues in the ones they've clicked on.

Once #439 lifts the validators into `stagebook/validate`, the extension can run them on every Stagebook file in the workspace via a palette command — equivalent to `npx stagebook validate '**/*'` from inside VS Code, with results landing in the Problems panel.

## Proposal

Add a command `stagebook.validateWorkspace` with palette entry "Stagebook: Validate Workspace". On invocation:

1. Glob the workspace for `**/*.stagebook.yaml` and `**/*.prompt.md` (respecting `.gitignore` and `files.exclude`).
2. Read each file and run the appropriate validator from `stagebook/validate`.
3. Convert each `Diagnostic` to a `vscode.Diagnostic` (the existing flow in [`extension.ts`](apps/vscode/src/extension.ts) already does this for the active file) and push to the language-diagnostic collection keyed by file URI.
4. Surface a summary in the status bar after the run: "Stagebook: N errors, M warnings across K files" with a click-through to the Problems panel.

## Acceptance

- Command palette has "Stagebook: Validate Workspace".
- Running it populates Problems with diagnostics from every Stagebook file in the workspace, not just opened ones.
- Diagnostics are byte-identical to what live-validation would produce on each file (same text, range, severity).
- Status bar shows aggregate counts after a run; clicking opens the Problems panel.

### PR DESCRIPTION
**Pull request by JamesPHoughton**
Fixes #442.

## What

Adds a command palette entry **"Stagebook: Validate Workspace"** (`stagebook.validateWorkspace`) that globs every `.stagebook.yaml` and `.prompt.md` file in the workspace, validates them, and populates the Problems panel — not just the files opened this session. A status-bar summary reports aggregate counts with click-through to the Problems panel.

Depends on the now-merged #439 (validators live in `stagebook/validate`).

## How

- **Refactor live validators into URI+source cores** (`validateTreatmentSourceAt` / `validatePromptSourceAt`). On-disk files and open editor buffers share one code path, so workspace-sweep diagnostics are **identical** to live validation (same `validateTreatmentWithDiff` + `checkFileReferences` for treatments, `validatePromptSource` for prompts — including the locale-consistency rule).
- **`checkFileReferences` is now awaitable + batched** (was fire-and-forget per-`stat`). The sweep awaits it, so the status-bar summary counts file-existence errors accurately instead of racing the stats. The stale-version guard is now applied once before the final `set` (strictly stronger than before — the old path-escape branch had no version guard).
- **Reads on-disk bytes, not `openTextDocument`, for unopened files.** Documents opened-but-never-shown can be disposed by VS Code, firing `onDidCloseTextDocument`, which would wipe the diagnostics we just set. When a file *is* open, its live buffer is preferred (respects unsaved edits).
- **Bounded concurrency** (`runPool`, cap 4) keeps the extension host responsive — each treatment validation is several seconds of synchronous Zod recursion.
- **Status-bar summary is scoped to the scanned files only** (not every diagnostic in the collection), so counts can't attribute errors to files outside the glob.

## Design decisions

- **`.gitignore` is intentionally not applied.** `findFiles` honors `files.exclude`/`search.exclude` but not `.gitignore`. This means coverage can differ from `npx stagebook validate '**/*'` if a treatment file is gitignored but not search-excluded. Chosen for simplicity (no new dependency, no nested-gitignore edge cases); revisit if it causes noise.

## Testing

- New unit tests (vitest, following the repo convention that pure logic lives in `src/lib/` and vscode-glue in `extension.ts` is verified in the Extension Development Host):
  - `runPool`: concurrency cap, failure isolation (rejected *and* synchronously-thrown thunks), empty list, `limit < 1`, `limit > task count`.
  - `diagnosticSummary`: count aggregation, singular/plural/mixed formatting, affected-vs-scanned tooltip distinction.
- Full local suite green: `npm test` (vitest, all workspaces) + `npm test -w stagebook-vscode` (103 tests), extension bundles via esbuild, Prettier clean.
- Pre-PR review by three subagents (correctness, test coverage, security). Correctness flagged the status-bar over-counting (fixed); security found no blocking issues and confirmed the `isWithinWorkspace` traversal guards survive the refactor on both the `loadImport` and asset-check paths.

## Known limitation (follow-up candidate)

A sweep only *sets* diagnostics on globbed files; it doesn't clear stale diagnostics for a Stagebook file deleted from disk that was never closed in the editor. The scoped status-bar count no longer mis-reports those, but the squiggle lingers (pre-existing behavior, same as live editing then deleting).

## Manual verification checklist (Extension Dev Host)

- [ ] Command appears in the palette as "Stagebook: Validate Workspace".
- [ ] Running it populates Problems for unopened files; counts match live validation when those files are then opened.
- [ ] Status bar shows aggregate counts; clicking opens the Problems panel.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by JamesPHoughton** [comment]
@codex review

**Comment by JamesPHoughton** [comment]
@codex review

File: apps/vscode/src/extension.ts [hunk 1]
```diff
@@ -311,7 +346,114 @@ function validatePromptFile(document: vscode.TextDocument): void {
...
+
+  await vscode.window.withProgress(
+    {
+      location: vscode.ProgressLocation.Window,
+      title: "Stagebook: validating workspace…",
+    },
+    async () => {
+      const [treatments, prompts] = await Promise.all([
+        vscode.workspace.findFiles("**/*.stagebook.yaml", "**/node_modules/**"),
+        vscode.workspace.findFiles("**/*.prompt.md", "**/node_modules/**"),
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts]: Good catch — this contradicted the stated design. Fixed in 6165e41: `findFiles` discards both `files.exclude` and `search.exclude` the moment a concrete exclude glob is passed, so I now merge `files.exclude` + `search.exclude` (+ `**/node_modules/**`) into a single exclude glob via a new `buildFindExcludeGlob` helper (unit-tested in `findExclude.test.ts`). Excluded files are now skipped as the user configured.

File: apps/vscode/src/extension.ts [hunk 2]
```diff
@@ -311,7 +346,114 @@ function validatePromptFile(document: vscode.TextDocument): void {
...
+        if (open) return open.getText();
+        return decoder.decode(await vscode.workspace.fs.readFile(uri));
+      };
+
+      const tasks: Array<() => Promise<void>> = [
+        ...treatments.map((uri) => async () => {
+          // The user explicitly asked to re-validate, so bypass the
+          // source-equality short-circuit for this run.
+          lastValidatedSource.delete(uri.toString());
+          await validateTreatmentSourceAt(uri, await readSource(uri));
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts]: Fixed in 6165e41. After the sweep I now drop the `lastValidatedSource` stamp for any scanned treatment that isn't currently open (no `onDidCloseTextDocument` fires for those, so the stamp would otherwise persist). A later open of an unchanged treatment will re-validate and pick up cross-file changes instead of hitting the source-equality shortcut. Open files keep their stamp since live editing manages it.

File: apps/vscode/src/extension.ts [hunk 3]
```diff
@@ -311,7 +346,114 @@ function validatePromptFile(document: vscode.TextDocument): void {
...
+        }),
+        ...prompts.map((uri) => async () => {
+          validatePromptSourceAt(uri, await readSource(uri));
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts]: Fixed in 6165e41. Both task thunks now wrap `readSource` in try/catch and call `setReadErrorDiagnostic(uri, e)` on failure, which sets a top-of-file error diagnostic with `source: "stagebook"`. Since `updateWorkspaceStatusBar` counts stagebook diagnostics on the scanned URIs, an unreadable file now shows as an error rather than a silent clean scan.
